### PR TITLE
Port to non-MacOS platforms

### PR DESCRIPTION
**Pull request by dvdkon**
This PR allows osxphotos to run on non-MacOS platforms (tested on NixOS Linux), both as a library and a CLI tool. This is accomplished mainly by gating specific functionality behind OS checks and providing alternative codepaths where necessary.

This mostly doesn't bring any complexity to the codebase, except for the need to gate new MacOS specific code and the need to deal with Unicode normalisation. MacOS does it in the background, other OSs can theoretically deal with any form of Unicode, but it seems Photos databases aren't internally consistent, so the best solution is to normalise on output.

Two tests are failing, `tests/test_cli.py::test_export_touch_files` and `test_export_touch_files_update`. I don't understand why only 31 files should be touched, so I can't fix the test. Other tests are either skipped or run fine.

A port to Linux will allow running tests on GitHub's CI platform, but that's not part of this PR.

### CODE REVIEW

**Comment by RhetTbull** [comment]
Wow -- thank you. This looks like a lot of work. As this is a big PR it will take me some time to review.

File: osxphotos/utils.py [hunk 1]
```diff
@@ -263,11 +279,13 @@ def list_photo_libraries():
     return lib_list
 
 
-def normalize_fs_path(path: str) -> str:
```
Comment (RhetTbull) [inline osxphotos/utils.py:266]: Changing unicode normalization from NFD to NFC will likely cause some file names to change which will corrupt existing exports using `--update`.  The switch should be gated to continue using NFD on macos. This was a deliberate choice as this is the default on HFS+ which is still common in the Mac ecosystem.

File: osxphotos/cli/__init__.py [hunk 2]
```diff
@@ -76,19 +74,24 @@
...
 from .tutorial import tutorial
-from .uuid import uuid
 from .version import version
 
+if is_macos:
+    from .add_locations import add_locations
+    from .batch_edit import batch_edit
+    from .import_cli import import_cli
+    from .photo_inspect import photo_inspect
+    from .query import query
```
Comment (RhetTbull) [inline osxphotos/cli/__init__.py:89]: Why is query gated here? query should be able to run on linux just like export with the exception of `--selected`.  The `--selected` option should also be gated in `cli_params.py` so it's not available on linux. (It is used by both export and query)

File: osxphotos/utils.py (hunk 1 above)
Comment (dvdkon) [reply osxphotos/utils.py:266]: I've changed it back to NFD for macOS. Sadly a lot of software behaves counter-intuitively when the Unicode form coming "from the keyboard" doesn't match the file, so I'm keeping it NFC on Linux/Windows/...

File: osxphotos/cli/__init__.py (hunk 2 above)
Comment (dvdkon) [reply osxphotos/cli/__init__.py]: For some reason I thought `query` relied more on `photoscript` than it does. Fixed. Thanks also for the reminder on `--selected`.

File: tests/test_cli.py [hunk 3]
```diff
@@ -72,12 +75,12 @@
...
-    "Frítest (3).jpg",
-    "Frítest_edited (1).jpeg",
-    "Frítest_edited.jpeg",
-    "Frítest.jpg",
+    "Frítest (1).jpg",
+    "Frítest (2).jpg",
+    "Frítest (3).jpg",
+    "Frítest_edited (1).jpeg",
+    "Frítest_edited.jpeg",
+    "Frítest.jpg",
```
Comment (RhetTbull) [inline tests/test_cli.py:83]: This causes most export tests to fail on macOS.  The unicode for macOS must be NFD or anyone who's previously exported unicode filenames with osxphotos will have their exports corrupted (duplicate exports, invalid updates, etc).  These file names are chosen because they came from users who had issues with exporting in unicode. For tests to pass on both macOS and linux you'll need separate versions of the test filenames. Can linux worth with NFD filenames? If so, perhaps it would be better to leave everything as NFD on export.

Comment (RhetTbull) [reply tests/test_cli.py:83]: Hi @dvdkon I would love to get this merged but need to get the unicode names in tests fixed. If you're able to add logic to the tests to use the right unicode names depending on platform I can get this merged. Perhaps it might be possible to transform the names from NFC/NFD when importing the test module so you don't need two separate sets of names.

Comment (dvdkon) [reply tests/test_cli.py:83]: Hi, sorry for not getting to this sooner. It would be best to leave Unicode issues to the user and continue as on macOS, but when running on Linux without the changes I've made, many tests fail, since somehow `osxphotos` is working with both NFC and NFD versions of the same strings. My guess is the Photos DB is not properly normalised, since it doesn't matter on HFS+.

I think your idea of re-normalising the filenames on import based on the platform is the best approach forward. It's going to be strict in requiring that it uses NFD on macOS and NFC elsewhere, but that might be for the better. I'll rebase on master and make the changes soon.

**Comment by dvdkon** [comment]
I got hold of a Mac and can confirm all tests now pass on macOS.

There's still a problem on Linux with two tests checking `touched date` counts. Apart from that, everything seems OK.

**Comment by RhetTbull** [comment]
Thanks for the update! I'll run the subset of tests that only run on my machine or require user interaction (timewarp, etc.) this weekend and try to get this merged.
